ffs/coreipfs/coreipfs.go

```diff
@@ -43,6 +43,29 @@ func (ci *CoreIpfs) Put(ctx context.Context, b blocks.Block) error {
 	return nil
 }
 
+// Remove removes a Cid from the Hot Storage.
+func (ci *CoreIpfs) Remove(ctx context.Context, c cid.Cid) error {
+	if err := ci.ipfs.Pin().Rm(ctx, path.IpfsPath(c), options.Pin.RmRecursive(true)); err != nil {
+		return fmt.Errorf("unpinning cid from ipfs node: %s", err)
+	}
+	return nil
+}
+
+// IsStored return if a particular Cid is stored.
+func (ci *CoreIpfs) IsStored(ctx context.Context, c cid.Cid) (bool, error) {
+	ci.ipfs.Block()
+	pins, err := ci.ipfs.Pin().Ls(ctx)
+	if err != nil {
+		return false, fmt.Errorf("getting pins from IPFS: %s", err)
+	}
+	for _, p := range pins {
```

Comment:
It could be worth keeping a cache of pins as a map/index? Pretty wild you have loop through the pin set.

Reply:
Sounds like a good idea since nobody else should be touching the IPFS node. 
Maybe only refreshing it after some time or similar. Will make an issue!